Techdebt refatorar modulo de adminclient do codex neodv 2150

henrique-lh opened the pull request:
## Description

With this pr:
- Add new admin module
- Improve code style
- Refactor python client to be the same entry point for all operations

## Related issues

- Closes: https://linear.app/datarisk/issue/NEODV-2150/techdebt-refatorar-modulo-de-adminclient-do-codex
- https://linear.app/datarisk/issue/NEODV-2149/techdebt-refatorar-modulo-de-treino-do-codex

## How to test it

* Set the API locally and change the `BASE_URL` in src/mlops_codex/constants.py
* Setup the codex locally

```bash
uv build
uv pip install -e .
```

Run the following script:

```python
from mlops_codex.administrator import DatariskMLops
from mlops_codex.train import CustomTrain

client = DatariskMLops(
    email="<email>",
    password="<password>",
    tenant="<tenant>"
)

client.list_groups()

group_token = client.create_group(name="codexGroup", description="Codex Group")

print("refresh group token")

new_group_token = client.refresh_group_token(name="codexGroup", force=True)

print('refresh ok')

e = client.train.setup_project_experiment(
    experiment_name='test',
    model_type='Classification',
    group='codexGroup',
)

print(e)

PATH = '<path_to_sample>/'

train_type = CustomTrain(
    training_reference='train_model',
    run_name='more test',
    python_version='3.10',
    input_data=PATH + 'dados.csv',
    source=PATH + 'app.py',
    requirements=PATH + 'requirements.txt',
)

print(train_type)

exe = client.train.run(
    experiment=e,
    train_type=train_type,
    wait_ready=True
)

print(exe)

print(exe.status)
```

henrique-lh committed:
- create boilerplate for new admin class

henrique-lh committed:
Merge branch 'master' into techdebt-refatorar-modulo-de-adminclient-do-codex-neodv-2150

# Conflicts:
#	README.md

henrique-lh committed:
- move from V2 folder to mlops

henrique-lh committed:
- move from V2 folder to mlops
- apply ruff

henrique-lh committed:
- add url enum

henrique-lh committed:
- add proxy to intercept services

henrique-lh committed:
- add new file for constants

henrique-lh committed:
- add attrs as necessary dependencies
- fix: warning: The `tool.uv.dev-dependencies` field (used in `pyproject.toml`) is deprecated and will be removed in a future release; use `dependency-groups.dev` instead

henrique-lh committed:
- update lock

ceciliacsilva commented on src/mlops_codex/constants.py:
```diff
@@ -0,0 +1 @@
+BASE_URL = 'https://neomaril.datarisk.net/api'
```
Is the codex only used outside of production in development? Just to confirm that the standard is being maintained.

ceciliacsilva commented on src/mlops_codex/administrator/proxy.py:
```diff
@@ -0,0 +1,29 @@
+
+
+@define(slots=True)
+class ServiceProxy:
```
The idea is to refactor the rest of the modules to use this?

henrique-lh committed:
- setup train module
- add pydantic

henrique-lh committed:
- inject headers

henrique-lh committed:
- add pydantic to project

henrique-lh committed:
- remove options

henrique-lh committed:
- move validations to utils folder

henrique-lh committed:
- update import from validations

henrique-lh committed:
- export interfaces to the user

henrique-lh committed:
- remove unnecessary field
- use f-strings

henrique-lh committed:
- update interfaces

henrique-lh committed:
- move validations to models

henrique-lh committed:
- ruff

henrique-lh committed:
- improve wait function
- ruff

henrique-lh committed:
- improve wait function
- ruff

henrique-lh committed:
- add data validation

henrique-lh committed:
- ruff

henrique-lh committed:
- ruff
- add str_to_path function to validate if a string is valid path to a file

henrique-lh committed:
- add custom exception

henrique-lh committed:
- rename file
- remove proxy and inject auth in the services

henrique-lh committed:
- remove print
- add header property

henrique-lh committed:
- remove proxy

henrique-lh committed:
- improve return of the wait function

henrique-lh committed:
- reorganize imports
- add docstrings to MLOpsExperiment
- add status property

henrique-lh committed:
- rename file
- move api calls to client.py
- turn setup_project_experiment into a class method
- remove kwargs to avoid user modify api call

henrique-lh committed:
- centralize client calls to API

henrique-lh committed:
- update Admin class name

henrique-lh committed:
- add docstrings
- ruff

henrique-lh committed:
- add isort to dev dependency group

henrique-lh committed:
- add external training option

henrique-lh committed:
- add docstrings

henrique-lh committed:
- add file extension validation

henrique-lh committed:
- add promote

henrique-lh committed:
- ruff

henrique-lh committed:
- add force option before create a new experiment

henrique-lh committed:
- ruff

henrique-lh committed:
- add describe and describe executions
- add load experiment and training execution feature
- ruff

henrique-lh committed:
- add download training execution feature
- ruff

henrique-lh committed:
- ruff

henrique-lh committed:
- update doc

henrique-lh committed:
- export automl interface
